### ISSUE TITLE
Startup with error "caught exception during start of feature 'Agency': getrandom"

### ISSUE THREAD

**Issue by Aaronchangji** [arangodb/arangodb#10261]
## My Environment

* __ArangoDB Version__:        3.5.1
* __Storage Engine__:             RocksDB
* __Deployment Mode__:       Cluster
* __Deployment Strategy__:   Manual Start & ArangoDB Starter
* __Configuration__:               
* __Infrastructure__:               own
* __Operating System__:        CentOS 6.8
* __Total RAM in your machine__:        128Gb
* __Disks in use__:        HDD
* __Used Package__:              

## Component, Query & Data

__Affected feature__: Installation

__AQL query (if applicable)__:

__AQL explain (if applicable)__:

__Dataset__:

__Size of your Dataset on disk__:

__Replication Factor & Number of Shards (Cluster only)__:

## Steps to reproduce

__Problem__:
I want to start arangodb with three machines for cluster mode. I firstly choose to use ArangoDB Starter with following command
```
arangodb --server.storage-engine=rocksdb --starter.data-dir=./ --starter.join w21,w22,w23
```                                                                 
on each machine(w21, w22, w23). And the log of w21 is
```
2019-10-16T11:11:01+08:00 |INFO| Starting arangodb version 0.14.12, build bbe2973 component=arangodb
2019-10-16T11:11:01+08:00 |INFO| ArangoDB Starter listening on 0.0.0.0:8528 (:8528) component=arangodb
2019-10-16T11:11:01+08:00 |INFO| Found ID from localhost peer component=arangodb
2019-10-16T11:11:01+08:00 |INFO| Found ID 'a93dddfd' from peer, looking for 'a93dddfd' component=arangodb
2019-10-16T11:11:01+08:00 |INFO| Using storage engine 'rocksdb' component=arangodb
2019-10-16T11:11:01+08:00 |INFO| Serving as master with ID 'a93dddfd' on :8528... component=arangodb
2019-10-16T11:11:01+08:00 |INFO| Waiting for 3 servers to show up.
 component=arangodb
2019-10-16T11:11:01+08:00 |INFO| Use the following commands to start other servers: component=arangodb

arangodb --starter.data-dir=./db2 --starter.join 127.0.0.1

arangodb --starter.data-dir=./db3 --starter.join 127.0.0.1

2019-10-16T11:11:01+08:00 |INFO| ArangoDB Starter listening on 0.0.0.0:8528 (:8528) component=arangodb
2019-10-16T11:11:27+08:00 |INFO| Added new peer 'ae77facd': 172.16.104.22, portOffset: 0 component=arangodb
2019-10-16T11:11:51+08:00 |INFO| Added new peer '2415c559': 172.16.104.23, portOffset: 0 component=arangodb
2019-10-16T11:11:52+08:00 |INFO| Starting service... component=arangodb
2019-10-16T11:11:52+08:00 |INFO| Looking for a running instance of agent on port 8531 component=arangodb
2019-10-16T11:11:52+08:00 |INFO| Starting agent on port 8531 component=arangodb
2019-10-16T11:11:53+08:00 |INFO| Looking for a running instance of dbserver on port 8530 component=arangodb
2019-10-16T11:11:53+08:00 |INFO| Starting dbserver on port 8530 component=arangodb
2019-10-16T11:11:54+08:00 |INFO| Looking for a running instance of coordinator on port 8529 component=arangodb
2019-10-16T11:11:54+08:00 |INFO| Starting coordinator on port 8529 component=arangodb
2019-10-16T11:12:02+08:00 |INFO| Failed to register upgrade plan changed callback component=arangodb error="context deadline exceeded"
2019-10-16T11:13:02+08:00 |INFO| Failed to get master URL, retrying in 5sec component=arangodb error="context deadline exceeded"
```
And I checked the log for agent:
```
2019-10-16T03:11:52Z [98974] INFO [e52b0] ArangoDB 3.5.1 [linux] 64bit, using jemalloc, build tags/v3.5.1-0-ged19ba500a, VPack 0.1.33, RocksDB 6.2.0, ICU 58.1, V8 7.1.302.28, OpenSSL 1.1.1d  10 Sep 2019
2019-10-16T03:11:52Z [98974] INFO [75ddc] detected operating system: Linux version 2.6.32-696.13.2.el6.x86_64 (mockbuild@c1bl.rdu2.centos.org) (gcc version 4.4.7 20120313 (Red Hat 4.4.7-18) (GCC) ) #1 SMP Thu Oct 5 21:22:16 UTC 2017
2019-10-16T03:11:52Z [98974] WARNING [e8b68] {memory} /sys/kernel/mm/transparent_hugepage/enabled is set to 'always'. It is recommended to set it to a value of 'never' or 'madvise'
2019-10-16T03:11:52Z [98974] WARNING [e8b68] {memory} /sys/kernel/mm/transparent_hugepage/defrag is set to 'always'. It is recommended to set it to a value of 'never' or 'madvise'
2019-10-16T03:11:52Z [98974] WARNING [f3108] {memory} execute 'sudo bash -c "echo madvise > /sys/kernel/mm/transparent_hugepage/enabled"'
2019-10-16T03:11:52Z [98974] WARNING [f3108] {memory} execute 'sudo bash -c "echo madvise > /sys/kernel/mm/transparent_hugepage/defrag"'
2019-10-16T03:11:52Z [98974] WARNING [3e451] {memory} It is recommended to set NUMA to interleaved.
2019-10-16T03:11:52Z [98974] WARNING [b25a4] {memory} put 'numactl --interleave=all' in front of your command
2019-10-16T03:11:52Z [98974] INFO [43396] {authentication} Jwt secret not specified, generating...
2019-10-16T03:11:52Z [98974] INFO [144fe] using storage engine rocksdb
2019-10-16T03:11:52Z [98974] INFO [3bb7d] {cluster} Starting up with role AGENT
2019-10-16T03:11:52Z [98974] INFO [a1c60] {syscall} file-descriptors (nofiles) hard limit is 65535, soft limit is 65535
2019-10-16T03:11:52Z [98974] INFO [3844e] {authentication} Authentication is turned off, authentication for unix sockets is turned on
2019-10-16T03:11:53Z [98974] INFO [e6460] created base application directory '/home/miao/arangodb/agent8531/apps/_db'
2019-10-16T03:11:53Z [98974] INFO [6ea38] using endpoint 'http+tcp://[::]:8531' for non-encrypted requests
2019-10-16T03:11:53Z [98974] INFO [cf3f4] ArangoDB (version 3.5.1 [linux]) is ready for business. Have fun!
2019-10-16T03:11:53Z [98974] ERROR [4ec19] {startup} startup aborted: caught exception during start of feature 'Agency': getrandom. shutting down
```

And then I use manual shart with following command to print more info on w21
```
numactl --interleave=all arangod --server.endpoint tcp://0.0.0.0:8531 \
    --agency.my-address tcp://172.16.104.21:8531 \
    --server.authentication false \
    --agency.activate true \
    --agency.size 3 \
    --agency.supervision true \
    --database.directory agent \
    --log.line-number true \
    --log.level DEBUG \
    --log.level startup=DEBUG \
    --log.level agency=DEBUG
```
And got message
```
2019-10-16T06:51:11Z [106160] A INFO [prepare@GreetingsFeature.cpp:36] [e52b0] ArangoDB 3.5.1 [linux] 64bit, using jemalloc, build tags/v3.5.1-0-ged19ba500a, VPack 0.1.33, RocksDB 6.2.0, ICU 58.1, V8 7.1.302.28, OpenSSL 1.1.1d  1[91/9889]
9
2019-10-16T06:51:11Z [106160] A DEBUG [prepare@ApplicationFeaturePhase.cpp:40] [12f47] {startup} Prepare for phase GreetingsPhase completed
2019-10-16T06:51:11Z [106160] A INFO [prepare@EnvironmentFeature.cpp:53] [75ddc] detected operating system: Linux version 2.6.32-696.13.2.el6.x86_64 (mockbuild@c1bl.rdu2.centos.org) (gcc version 4.4.7 20120313 (Red Hat 4.4.7-18) (GCC) ) #
1 SMP Thu Oct 5 21:22:16 UTC 2017
2019-10-16T06:51:11Z [106160] A DEBUG [prepare@EnvironmentFeature.cpp:350] [63a7a] host ASLR is in use for shared libraries, stack, mmap, VDSO, heap and memory managed through brk()
2019-10-16T06:51:11Z [106160] A DEBUG [prepare@ApplicationFeaturePhase.cpp:40] [12f47] {startup} Prepare for phase BasicsPhase completed
2019-10-16T06:51:11Z [106160] A DEBUG [prepare@AuthenticationFeature.cpp:164] [713c0] {authentication} Not creating user manager
2019-10-16T06:51:11Z [106160] A INFO [prepare@AuthenticationFeature.cpp:172] [43396] {authentication} Jwt secret not specified, generating...
2019-10-16T06:51:11Z [106160] A DEBUG [setJwtSecret@TokenCache.cpp:69] [71a76] {authentication} Setting jwt secret of size 64
2019-10-16T06:51:11Z [106160] A DEBUG [prepare@EngineSelectorFeature.cpp:63] [98b5c] {startup} looking for previously selected engine in file '/home/miao/arangodb/agent/ENGINE'
2019-10-16T06:51:11Z [106160] A INFO [prepare@EngineSelectorFeature.cpp:123] [144fe] using storage engine rocksdb
2019-10-16T06:51:11Z [106160] A DEBUG [prepare@ApplicationFeaturePhase.cpp:40] [12f47] {startup} Prepare for phase DatabasePhase completed
2019-10-16T06:51:11Z [106160] A INFO [reportRole@ClusterFeature.cpp:346] [3bb7d] {cluster} Starting up with role AGENT
2019-10-16T06:51:11Z [106160] A DEBUG [prepare@ApplicationFeaturePhase.cpp:40] [12f47] {startup} Prepare for phase ClusterPhase completed
2019-10-16T06:51:11Z [106160] A DEBUG [prepare@ApplicationFeaturePhase.cpp:40] [12f47] {startup} Prepare for phase V8Phase completed
2019-10-16T06:51:11Z [106160] A DEBUG [prepare@ApplicationFeaturePhase.cpp:40] [12f47] {startup} Prepare for phase AQLPhase completed
2019-10-16T06:51:11Z [106160] A DEBUG [prepare@ApplicationFeaturePhase.cpp:40] [12f47] {startup} Prepare for phase ServerPhase completed
2019-10-16T06:51:11Z [106160] A DEBUG [prepare@ApplicationFeaturePhase.cpp:40] [12f47] {startup} Prepare for phase FoxxPhase completed
2019-10-16T06:51:11Z [106160] A DEBUG [prepare@AgencyFeature.cpp:290] [693a2] {agency} Agency endpoint tcp://172.16.104.21:8531
2019-10-16T06:51:11Z [106160] A DEBUG [prepare@ApplicationFeaturePhase.cpp:40] [12f47] {startup} Prepare for phase AgencyPhase completed
2019-10-16T06:51:11Z [106160] A DEBUG [prepare@ApplicationFeaturePhase.cpp:40] [12f47] {startup} Prepare for phase FinalPhase completed
2019-10-16T06:51:11Z [106160] A DEBUG [start@ApplicationFeaturePhase.cpp:44] [d730b] {startup} Start for phase CommunicationPhase completed
2019-10-16T06:51:11Z [106160] A DEBUG [start@ApplicationFeaturePhase.cpp:44] [d730b] {startup} Start for phase GreetingsPhase completed
2019-10-16T06:51:11Z [106160] A INFO [start@FileDescriptorsFeature.cpp:86] [a1c60] {syscall} file-descriptors (nofiles) hard limit is 65535, soft limit is 65535
2019-10-16T06:51:11Z [106160] A DEBUG [setLocale@LanguageFeature.cpp:64] [f6e04] {config} using default language 'en'
2019-10-16T06:51:11Z [106160] A DEBUG [start@SchedulerFeature.cpp:191] [14e6f] {startup} scheduler has started
2019-10-16T06:51:11Z [106160] A DEBUG [start@ApplicationFeaturePhase.cpp:44] [d730b] {startup} Start for phase BasicsPhase completed
2019-10-16T06:51:11Z [106160] A INFO [start@AuthenticationFeature.cpp:215] [3844e] {authentication} Authentication is turned off, authentication for unix sockets is turned on
2019-10-16T06:51:11Z [106160] A DEBUG [setLocale@LanguageFeature.cpp:64] [f6e04] {config} using default language 'en'
2019-10-16T06:51:11Z [106160] A WARNING [TRI_VerifyLockFile@files.cpp:1436] [b387d] found existing lockfile '/home/miao/arangodb/agent/LOCK' of previous process with pid 74033, but that process seems to be dead already
2019-10-16T06:51:11Z [106160] A DEBUG [start@RocksDBEngine.cpp:666] [528b8] {startup} found existing column families: default, Documents, PrimaryIndex, EdgeIndex, VPackIndex, GeoIndex, FulltextIndex
2019-10-16T06:51:12Z [106160] A DEBUG [start@ApplicationFeaturePhase.cpp:44] [d730b] {startup} Start for phase DatabasePhase completed
2019-10-16T06:51:12Z [106160] A DEBUG [start@ApplicationFeaturePhase.cpp:44] [d730b] {startup} Start for phase ClusterPhase completed
2019-10-16T06:51:12Z [106160] A DEBUG [start@ApplicationFeaturePhase.cpp:44] [d730b] {startup} Start for phase V8Phase completed
2019-10-16T06:51:12Z [106160] A DEBUG [start@ApplicationFeaturePhase.cpp:44] [d730b] {startup} Start for phase AQLPhase completed
2019-10-16T06:51:12Z [106160] A DEBUG [check@Version.cpp:100] [3da0f] {startup} found VERSION file '/home/miao/arangodb/agent/VERSION-1', content: {"version":30501,"tasks":{}}\n
2019-10-16T06:51:12Z [106160] A DEBUG [check@Version.cpp:142] [e9cc3] {startup} version match: last version 30501, current version 30501
2019-10-16T06:51:12Z [106160] A DEBUG [runTasks@Upgrade.cpp:295] [cc057] {startup} Upgrade: cluster mismatch, skipping createSystemCollectionsAndIndices
2019-10-16T06:51:12Z [106160] A DEBUG [runTasks@Upgrade.cpp:295] [cc057] {startup} Upgrade: cluster mismatch, skipping createSystemStatisticsDBServer
2019-10-16T06:51:12Z [106160] A DEBUG [runTasks@Upgrade.cpp:288] [fd4e0] {startup} Upgrade: DB system, Skipping addDefaultUserOther
2019-10-16T06:51:12Z [106160] A DEBUG [runTasks@Upgrade.cpp:295] [cc057] {startup} Upgrade: cluster mismatch, skipping persistLocalDocumentIds
2019-10-16T06:51:12Z [106160] A DEBUG [runTasks@Upgrade.cpp:295] [cc057] {startup} Upgrade: cluster mismatch, skipping renameReplicationApplierStateFiles
2019-10-16T06:51:12Z [106160] A DEBUG [runTasks@Upgrade.cpp:295] [cc057] {startup} Upgrade: cluster mismatch, skipping dropLegacyAnalyzersCollection
2019-10-16T06:51:12Z [106160] A DEBUG [runTasks@Upgrade.cpp:359] [e5a77] {startup} Upgrade: Writing VERSION file
2019-10-16T06:51:12Z [106160] A INFO [dump@EndpointList.cpp:218] [6ea38] using endpoint 'http+tcp://0.0.0.0:8531' for non-encrypted requests
2019-10-16T06:51:12Z [106160] A DEBUG [startListening@GeneralServer.cpp:86] [dc45a] bound to endpoint 'http+tcp://0.0.0.0:8531'
2019-10-16T06:51:12Z [106160] A DEBUG [start@ApplicationFeaturePhase.cpp:44] [d730b] {startup} Start for phase ServerPhase completed
2019-10-16T06:51:12Z [106160] A INFO [start@BootstrapFeature.cpp:334] [cf3f4] ArangoDB (version 3.5.1 [linux]) is ready for business. Have fun!
2019-10-16T06:51:12Z [106160] A DEBUG [start@ApplicationFeaturePhase.cpp:44] [d730b] {startup} Start for phase FoxxPhase completed
2019-10-16T06:51:12Z [106160] A DEBUG [start@AgencyFeature.cpp:308] [a77c8] {agency} Starting agency personality
2019-10-16T06:51:12Z [106160] A DEBUG [start@Agent.cpp:135] [9a90e] {agency} Starting agency comm worker.
2019-10-16T06:51:12Z [106160] A DEBUG [start@AgencyFeature.cpp:311] [b481d] {agency} Loading agency
2019-10-16T06:51:12Z [106160] A DEBUG [load@Agent.cpp:807] [c07e1] {agency} Loading persistent state.
2019-10-16T06:51:12Z [106160] A DEBUG [loadOrPersistConfiguration@State.cpp:949] [a27cb] {agency} New agency!
2019-10-16T06:51:12Z [106160] A ERROR [start@ApplicationServer.cpp:683] [4ec19] {startup} startup aborted: caught exception during start of feature 'Agency': getrandom. shutting down
```

Is there any hint for this problem? Thanks

**Comment by OmarAyo** [comment]
Hi @Aaronchangji 

Where you able to sort out the issue, are you now up and running

If not please do the followings, so we can start troubleshotting:
- use the following Starter commands:
```
machine1: arangodb --starter.data-dir=./node1 --starter.join w21,w22,w23 --log.verbose
machine2: arangodb --starter.data-dir=./node2 --starter.join w21,w22,w23 --log.verbose
machine3: arangodb --starter.data-dir=./node3 --starter.join w21,w22,w23 --log.verbose
```
This command will create all the cluster components sub-directories 
e.g.
```
agent1: ./node1/agent8531
coordinator1: ./node1/coordinator8529
```
- increase the log verbosity to `DEBUG` for each node. This can be done by editing the `
arangod.conf` file that in this case is located under the path `./node1/agent8541/arangod.conf` - just change the following section:
```
[log]
level = INFO //change this to DEBUG
```
- repeat the above step to all of the cluster componet (Coordinators and DBServers) in all nodes (node1, node2, node3)
e.g.
1. go to `./node2/coordinator8539/arangod.conf`
2. change:
```
[log]
level = INFO //change this to DEBUG
```
Once this is done and error occurs, please attach all the log files (arangodb.log and arangod.log) from all the cluster components (agents, dbservers, coordinators) from all 3 nodes

Best

**Comment by Aaronchangji** [comment]
Here are the log files.
[arangod_node1_agent.log](https://github.com/arangodb/arangodb/files/3762267/arangod_node1_agent.log)
[arangod_node1_coordinator.log](https://github.com/arangodb/arangodb/files/3762256/arangod_node1_coordinator.log)
[arangod_node1_dbserver.log](https://github.com/arangodb/arangodb/files/3762257/arangod_node1_dbserver.log)

[arangod_node2_agent.log](https://github.com/arangodb/arangodb/files/3762258/arangod_node2_agent.log)
[arangod_node2_coordinator.log](https://github.com/arangodb/arangodb/files/3762259/arangod_node2_coordinator.log)
[arangod_node2_dbserver.log](https://github.com/arangodb/arangodb/files/3762260/arangod_node2_dbserver.log)

[arangod_node3_agent.log](https://github.com/arangodb/arangodb/files/3762261/arangod_node3_agent.log)
[arangod_node3_coordinator.log](https://github.com/arangodb/arangodb/files/3762262/arangod_node3_coordinator.log)
[arangod_node3_dbserver.log](https://github.com/arangodb/arangodb/files/3762263/arangod_node3_dbserver.log)

[arangodb_node1.log](https://github.com/arangodb/arangodb/files/3762264/arangodb_node1.log)
[arangodb_node2.log](https://github.com/arangodb/arangodb/files/3762265/arangodb_node2.log)
[arangodb_node3.log](https://github.com/arangodb/arangodb/files/3762266/arangodb_node3.log)

It seems the log files do not provide too much information. And the log provided last time seems to contain more messages
```
2019-10-16T06:51:12Z [106160] A DEBUG [start@ApplicationFeaturePhase.cpp:44] [d730b] {startup} Start for phase ServerPhase completed
2019-10-16T06:51:12Z [106160] A INFO [start@BootstrapFeature.cpp:334] [cf3f4] ArangoDB (version 3.5.1 [linux]) is ready for business. Have fun!
2019-10-16T06:51:12Z [106160] A DEBUG [start@ApplicationFeaturePhase.cpp:44] [d730b] {startup} Start for phase FoxxPhase completed
2019-10-16T06:51:12Z [106160] A DEBUG [start@AgencyFeature.cpp:308] [a77c8] {agency} Starting agency personality
2019-10-16T06:51:12Z [106160] A DEBUG [start@Agent.cpp:135] [9a90e] {agency} Starting agency comm worker.
2019-10-16T06:51:12Z [106160] A DEBUG [start@AgencyFeature.cpp:311] [b481d] {agency} Loading agency
2019-10-16T06:51:12Z [106160] A DEBUG [load@Agent.cpp:807] [c07e1] {agency} Loading persistent state.
2019-10-16T06:51:12Z [106160] A DEBUG [loadOrPersistConfiguration@State.cpp:949] [a27cb] {agency} New agency!
2019-10-16T06:51:12Z [106160] A ERROR [start@ApplicationServer.cpp:683] [4ec19] {startup} startup aborted: caught exception during start of feature 'Agency': getrandom. shutting down
```

**Comment by cckn1ght** [comment]
Any updates on this issue? I have faced the same issue when manually creating an agent using docker. 
My command is: 
```
docker run -it --rm -e ARANGO_NO_AUTH=1 --network=host \
  arangodb:3.5.3 arangod --server.endpoint tcp://tw-node42:8531 \
  --agency.my-address tcp://tw-node42:8531 \
  --server.authentication false   --agency.activate true \
  --agency.size 3   --agency.supervision true \
  --database.directory agent --log.level debug
```
The log is:

```
cannot start with NUMA numactl --interleave=all: please ensure that docker is running with --cap-add SYS_NICE
automatically choosing storage engine
Initializing database...Hang on...
Database initialized...Starting System...
{config} using user supplied config file '/tmp/arangod.conf'
2019-12-09T09:54:34Z [1] INFO [e52b0] ArangoDB 3.5.3 [linux] 64bit, using jemalloc, build tags/v3.5.3-0-gf9ff700153, VPack 0.1.33, RocksDB 6.2.0, ICU 58.1, V8 7.1.302.28, OpenSSL 1.1.1d  10 Sep 2019
2019-12-09T09:54:34Z [1] INFO [75ddc] detected operating system: Linux version 3.10.0-327.el7.x86_64 (builder@kbuilder.dev.centos.org) (gcc version 4.8.3 20140911 (Red Hat 4.8.3-9) (GCC) ) #1 SMP Thu Nov 19 22:10:57 UTC 2015
2019-12-09T09:54:34Z [1] WARNING [118b0] {memory} maximum number of memory mappings per process is 262144, which seems too low. it is recommended to set it to at least 2048000
2019-12-09T09:54:34Z [1] WARNING [49528] {memory} execute 'sudo sysctl -w "vm.max_map_count=2048000"'
2019-12-09T09:54:34Z [1] WARNING [7a7af] {performance} /proc/sys/vm/zone_reclaim_mode is set to '1'. It is recommended to set it to a value of 0
2019-12-09T09:54:34Z [1] WARNING [11b2b] {performance} execute 'sudo bash -c "echo 0 > /proc/sys/vm/zone_reclaim_mode"'
2019-12-09T09:54:34Z [1] WARNING [3e451] {memory} It is recommended to set NUMA to interleaved.
2019-12-09T09:54:34Z [1] WARNING [b25a4] {memory} put 'numactl --interleave=all' in front of your command
2019-12-09T09:54:34Z [1] DEBUG [63a7a] host ASLR is in use for shared libraries, stack, mmap, VDSO, heap and memory managed through brk()
2019-12-09T09:54:34Z [1] DEBUG [713c0] {authentication} Not creating user manager
2019-12-09T09:54:34Z [1] INFO [43396] {authentication} Jwt secret not specified, generating...
2019-12-09T09:54:34Z [1] DEBUG [71a76] {authentication} Setting jwt secret of size 64
2019-12-09T09:54:34Z [1] INFO [144fe] using storage engine rocksdb
2019-12-09T09:54:34Z [1] INFO [3bb7d] {cluster} Starting up with role AGENT
2019-12-09T09:54:34Z [1] INFO [a1c60] {syscall} file-descriptors (nofiles) hard limit is 1048576, soft limit is 1048576
2019-12-09T09:54:34Z [1] DEBUG [f6e04] {config} using default language 'en_US'
2019-12-09T09:54:34Z [1] INFO [24783] created database directory 'agent'
2019-12-09T09:54:34Z [1] INFO [3844e] {authentication} Authentication is turned off, authentication for unix sockets is turned on
2019-12-09T09:54:34Z [1] DEBUG [08f3c] {config} Writing default language to file 'agent/LANGUAGE'
2019-12-09T09:54:34Z [1] DEBUG [f6e04] {config} using default language 'en_US'
2019-12-09T09:54:35Z [1] DEBUG [f6cbd] Writing server id to file 'agent/SERVER'
2019-12-09T09:54:35Z [1] INFO [6ea38] using endpoint 'http+tcp://tw-node42:8531' for non-encrypted requests
2019-12-09T09:54:35Z [1] DEBUG [dc45a] bound to endpoint 'http+tcp://tw-node42:8531'
2019-12-09T09:54:35Z [1] INFO [cf3f4] ArangoDB (version 3.5.3 [linux]) is ready for business. Have fun!
2019-12-09T09:54:35Z [1] ERROR [4ec19] {startup} startup aborted: caught exception during start of feature 'Agency': getrandom. shutting down
2019-12-09T09:54:35Z [1] DEBUG [74339] accept failed: Operation canceled
```
*Only docker image of version arangdb:3.5[.3] yields this error. arangodb:3.4.8 works fine.*

**Comment by cckn1ght** [comment]
### My Environment
ArangoDB Version: 3.5.3
Storage Engine: RocksDB
Deployment Mode: Cluster with Docker
Deployment Strategy: Manual Start & ArangoDB Starter
Configuration:
Infrastructure: own
Operating System: CentOS 7
Total RAM in your machine: 128Gb
Disks in use: HDD
Used Package:

**Comment by jsteemann** [comment]
Hi @Aaronchangji : there has been a bugfix PR for this recently:
https://github.com/arangodb/arangodb/pull/10585

I can't tell right now whether it fixes the exact same problem, but it was related to calls of `getrandom` at startup. So far we were not aware the problem happens on CentOS 7 as well. We were only aware of it happening on CentOS 6.
Anyway, the PR hopefully fixes it, because it will avoid an underlying third party library to call getrandom.

The PR has been merged into the 3.5 branch, but there hasn't been a 3.5 release including the fix. The next 3.5 release that will include the fix will be 3.5.4.

**Comment by maxkernbach** [comment]
Hi @Aaronchangji,

Version 3.5.4, which includes the previously mentioned fix, has just been released.
Could you upgrade your deployment and check if the error persists?

Please find the latest release in the [download section](https://www.arangodb.com/download/).

**Comment by Simran-B** [comment]
Hi @Aaronchangji,
we haven't heard from you back for quite a while. Please upgrade to the latest version of 3.5 (or consider even 3.6) and see if it's still a problem. Closing this ticket for now.
